tests/platform_tests/sfp/test_sfputil.py

```diff
@@ -122,54 +404,50 @@ def test_check_sfputil_eeprom(duthosts, enum_rand_one_per_hwsku_frontend_hostnam
-    if duthost.facts["asic_type"] == "vs" and sfp_presence['rc'] == 2:
-        pass
-    else:
-        assert sfp_presence['rc'] == 0, "Run command '{}' failed".format(cmd_sfp_presence)
-
-    parsed_presence = parse_output(sfp_presence["stdout_lines"][2:])
-    for intf in dev_conn:
-        if intf not in xcvr_skip_list[duthost.hostname]:
-            assert intf in parsed_presence, "Interface is not in output of '{}'".format(cmd_sfp_presence)
-            assert parsed_presence[intf] == "Present", "Interface presence is not 'Present'"
-
-    logging.info("Check interface status")
-    mg_facts = duthost.get_extended_minigraph_facts(tbinfo)
-    intf_facts = duthost.interface_facts(up_ports=mg_facts["minigraph_ports"])["ansible_facts"]
-    assert len(intf_facts["ansible_interface_link_down_ports"]) == 0, \
-        "Some interfaces are down: {}".format(intf_facts["ansible_interface_link_down_ports"])
+    phy_intfs_to_test_per_asic = get_phy_intfs_to_test_per_asic(duthost,
+                                                                conn_graph_facts,
+                                                                enum_frontend_asic_index,
+                                                                xcvr_skip_list)
+    for phy_intf, logical_intfs_list in phy_intfs_to_test_per_asic.items():
+        logical_intf = logical_intfs_list[0]
+        with DisablePhysicalInterface(duthost, enum_frontend_asic_index, phy_intf, logical_intfs_list):
+            cmd_sfp_presence_per_intf = cmd_sfp_presence + " -p {}".format(logical_intf)
+
+            cmd_sfp_reset_intf = "{} {}".format(cmd_sfp_reset, logical_intf)
+            logging.info("resetting {} physical interface {}".format(logical_intf, phy_intf))
+            reset_result = duthost.command(cmd_sfp_reset_intf)
+            assert reset_result["rc"] == 0, "'{}' failed".format(cmd_sfp_reset_intf)
+            time.sleep(I2C_WAIT_TIME_AFTER_SFP_RESET)
+
+            logging.info("Check sfp presence again after reset")
+            sfp_presence = duthost.command(cmd_sfp_presence_per_intf, module_ignore_errors=True)
+
+            # For vs testbed, we will get expected Error code `ERROR_CHASSIS_LOAD = 2` here.
+            if duthost.facts["asic_type"] == "vs" and sfp_presence['rc'] == 2:
+                pass
+            else:
+                assert sfp_presence['rc'] == 0, \
+                    "Run command '{}' failed".format(cmd_sfp_presence_per_intf)
+
+            parsed_presence = parse_output(sfp_presence["stdout_lines"][2:])
+            assert logical_intf in parsed_presence, \
+                "Interface is not in output of '{}'".format(cmd_sfp_presence_per_intf)
+            assert parsed_presence[logical_intf] == "Present", \
+                "Interface presence is not 'Present' for {}".format(logical_intf)
+
+    # Check interface status for all interfaces in the end just in case
+    assert check_interface_status(duthost,
```

Comment:
@longhuan-cisco Can we deinitialize all lanes (similar to what we do via set_datapath_deinit API) after issuing sfputil reset?
This is needed to ensure that when the module transitions to high power mode, the modules does not initialize itself with the default settings. Hence, this will give software the control of initializing the module after reset is issued and prevent ports from linking up just after module is brought in high power mode.

Reply:
Were you talking about adding something like set_datapath_deinit on cmis modules right after reset, to avoid the potential early link-up (on certain module) due to manually toggling to high power mode after reset?

Since there's still question on whether to manually toggle lpmode after reset for cmis, will wait after closure on discussion https://github.com/sonic-net/sonic-mgmt/pull/13108/files#r1785319428

Reply:
@longhuan-cisco I have responded to the comment now. Let me know if you still have further questions.

Reply:
Updated as discussed over email.